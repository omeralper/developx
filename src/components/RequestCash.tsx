import React from "react";
import { requestCashAction, requestCashActionFunction } from "../actions/CashActions";
import { ICashResponse } from "../api/CashApi";
import { connect } from "react-redux";

export interface IRequestCashProps {
    requestCashAction: requestCashActionFunction;
}

interface IRequestCashState {
    cashAmount: number;
}

export class RequestCash extends React.PureComponent<IRequestCashProps,IRequestCashState> {
    constructor(props: IRequestCashProps) {
        super(props);
        this.state = {
            cashAmount: 0
        };
    }

    onCashSubmit = () => { 
        this.props.requestCashAction(this.state.cashAmount);
    }

    setCashAmount = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({
            cashAmount: +event.target.value
         })  
    }

    render() {
        return (
            <div className="RequestCash">
               <form className="text-center border border-light p-5">
                    <p className="h4 mb-4">Withdraw Cash</p>
                    <input type="number" className="form-control mb-4" placeholder="Amount" onChange={this.setCashAmount} />
                    <input type="buton" className="btn btn-info my-4" onClick={this.onCashSubmit} value="Submit"/>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = {
    requestCashAction
}

export default connect(null,mapDispatchToProps)(RequestCash);