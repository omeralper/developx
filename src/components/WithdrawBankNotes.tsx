import React from "react";
import { ICashResponseEntity } from "../api/CashApi";
import { connect } from "react-redux";
import { ICashReducerState } from "../reducers/CashReducer";

export interface IWithdrawBankNotesProps {
    withdrawMoneySet: ICashResponseEntity[],
    notAvailable: boolean
}

export const WithdrawBankNotes: React.StatelessComponent<IWithdrawBankNotesProps> = ({
    withdrawMoneySet,
    notAvailable
}) => {
    return (
        <div className="WithdrawBankNotes">
         {notAvailable ? "Not Available" : 
            <ul>
                {withdrawMoneySet.map((cashEntity, i) => (
                    <li key={i}>Bank note {cashEntity.bankNote} - Amount  {cashEntity.amount}</li>
                ))}
            </ul>
         }
        </div>
    );
}

const mapStateToProps = ({withdrawMoneySet,notAvailable}: ICashReducerState) => ({
    withdrawMoneySet,
    notAvailable
})

export default connect(mapStateToProps)(WithdrawBankNotes);