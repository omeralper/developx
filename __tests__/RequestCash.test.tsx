import * as React from "react";
import * as renderer from "react-test-renderer";
import {RequestCash, IRequestCashProps } from "../src/components/RequestCash";
import { shallow } from "enzyme";

let props = {} as IRequestCashProps;

describe("Request Cash  Component", () => {
    beforeEach(() =>{
        props = {} as IRequestCashProps;
    });

    it("shoud render withdrawbanknotes component properly", () => {
        const component = renderer.create(<RequestCash {...props} />);
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("shoud call request cash actions on submit", () => {
        props.requestCashAction = jest.fn();
        const wrapper = shallow(<RequestCash {...props} />);
        const button = wrapper.find(".btn-info");
        button.simulate("click");
        expect((props.requestCashAction as jest.Mock).mock.calls.length).toBe(1);
    });
})