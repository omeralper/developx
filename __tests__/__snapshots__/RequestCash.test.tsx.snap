// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Cash  Component shoud render withdrawbanknotes component properly 1`] = `
<div
  className="RequestCash"
>
  <form
    className="text-center border border-light p-5"
  >
    <p
      className="h4 mb-4"
    >
      Withdraw Cash
    </p>
    <input
      className="form-control mb-4"
      onChange={[Function]}
      placeholder="Amount"
      type="number"
    />
    <input
      className="btn btn-info my-4"
      onClick={[Function]}
      type="buton"
      value="Submit"
    />
  </form>
</div>
`;
